### ISSUE TITLE
Locale issues

### ISSUE THREAD

**Issue by Pyrohh** [neovim/neovim#2393]
refs #2386

It doesn't seem like Nvim ([or libtermkey?](https://github.com/neovim/neovim/issues/2386#issuecomment-91361973)) follows POSIX 2013 regarding environment variables, as [said here](https://github.com/neovim/neovim/issues/2386#issuecomment-91359820) by @bfredl:

> Looking at the docs it looks like the order of precendence should be LC_ALL > LC_CTYPE > LANG , and yes LC_ALL=C does break vim but doesn't break nvim. I thoght that the C library (or some other system library) was handling the interpretation of all these variables? (At least, every application handling all these itself sounds quite fragile and duplicated work)

**Comment by bfredl** [comment]
moving my comment over here:

The glibc manual claims this is the POSIX way:
```
#include <locale.h>
#include <langinfo.h>
int main() {
    setlocale(LC_ALL, "");
    printf("%s\n",nl_langinfo(CODESET));
}
```
where `LC_ALL` means "take all environment variables into account" and not just `LC_ALL` . It seem to use the `LC_ALL > LC_CTYPE > LANG` order, at least.

**Comment by splinterofchaos** [comment]
@leonerd, is there a reason libtermkey works this way?

**Comment by bfredl** [comment]
And I don't understand libtermkey's notion of "raw" and "utf8" terminals? What if the terminal is latin-1 ? (which IIRC is the default on PuTTY, for instance )

**Comment by bfredl** [comment]
I think I see, libtermkey needs to just special case for the variable-length nature of utf8 (and leaves decoding to the user). But if `nvim` is already reading the locale elsewhere (for messages etc), it could just pass this information (utf8 or not) to libtermkey as a flag to `termkey_new`.

**Comment by bfredl** [comment]
Anyone that have had issues might test #2469, but please remove any work-around environment variables before this, as the whole point is that it should work Out of the Box™

**Comment by bfredl** [comment]
closed by #2469
